src/shogun/optimization/FirstOrderSAGCostFunctionInterface.h

```diff
@@ -0,0 +1,161 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Elfarouk
+ */
+
+#ifndef FIRSTORDERSAGCOSTFUNCTIONINTERFACE_H
+#define FIRSTORDERSAGCOSTFUNCTIONINTERFACE_H
+
+#include <stan/math.hpp>
+#include <shogun/mathematics/eigen3.h>
+#include <shogun/lib/config.h>
+#include <shogun/optimization/FirstOrderSAGCostFunction.h>
+#include <functional>
+#include <vector>
+#include <shogun/lib/SGMatrix.h>
+#include <shogun/lib/SGVector.h>
+
+using Eigen::Dynamic;
+using Eigen::Matrix;
+using stan::math::var;
+using std::function;
```

Comment:
It does not feel good to have type aliases in headers. They also become effective in places where the header is included, no?

Reply:
second that this actually contaminates any file that will include this header

Reply:
I thought that if you use using std::function; in a header, then you'd need to access it from another file which has the include as FirstOrderSAGCostFunctionInterface::function no? In other words, shouldn't it be under the scope of the .h file?

Reply:
Actually you're right, I just forgot my c++! Including it in the header file does include it in other files as well. I'll get rid of it.